Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -10984,17 +11012,47 @@ def events_error_handler(res):
         demisto.error(header_msg + api_call_info)
         raise DemistoException(header_msg + error, DemistoException)
 
-    zipped_data = gzip.compress(data.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
     client = BaseClient(base_url=xsiam_url)
+    data_chunks = split_data_to_chunks(data, XSIAM_EVENT_CHUNK_SIZE)
+    amount_of_events = 0
+    for data_chunk in data_chunks:
+        amount_of_events += len(data_chunk)
+        data_chunk = '\n'.join(data_chunk)
+        zipped_data = gzip.compress(data_chunk.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
+        send_to_xsiam_with_retries(client, events_error_handler, header_msg, headers, num_of_attempts, xsiam_url,
+                                   zipped_data)
+    demisto.updateModuleHealth({'eventsPulled': amount_of_events})
+
 
+def send_to_xsiam_with_retries(client, events_error_handler, header_msg, headers, num_of_attempts, xsiam_url,
```

Comment:
I think this doesn't need to be its own function (especially because it has a big signature that just gets passed to another function as is). Perhaps we can add this functionality via an optional argument to the existing `xsiam_api_call_with_retries()` e.g. `return_type='json'`, which will also add an implicit error handling - I'm not 100% sure that's the right way to go either.

I'll leave it up to you to decide how to handle this